src/main/kotlin/lotto/data/WinningLotto.kt

```diff
@@ -1,3 +1,24 @@
 package lotto.data
 
-data class WinningLotto(val lotto: Lotto, val bonusNumber: LottoNumber)
+data class WinningLotto(val lotto: Lotto, val bonusNumber: LottoNumber) {
+
+    init {
+        validateWinningLotto()
+    }
+
+    fun countMatchingNumbers(lotto: Lotto): Int {
+        return this.lotto.selectNumbers.intersect(lotto.selectNumbers).size
```

Comment:
모든 로직을 작성할때 dot(.)이 몇 개인지 체크해보는것도 좋을 것 같아요. 개인적으로는 Stream API와 같이 중간 오퍼레이터들이 있는 지연 연산되는 특수한 경우를 제외하고는 2개 이상의 dot(.)이 있는경우 내가 지금 이 객체의 속살을 모두 꺼내서 요리를 하는게 아닌가? 고민해볼 수 있어요. 

```suggestion
        return this.lotto.matchCount(targetLotto)
```